lib/ansible/plugins/inventory/__init__.py

```diff
@@ -423,6 +423,10 @@ def _add_host_to_keyed_groups(self, keys, variables, host, strict=False):
                             self.inventory.add_child(gname, host)
 
                             if raw_parent_name:
+                                try:
+                                    raw_parent_name = self._compose(raw_parent_name, variables)
+                                except Exception as e:
+                                    pass
```

Comment:
-1 since a misspelled variable automatically becomes a string identifier, we already went through hell because of this on the original `with_` implementation and that is why we mandated `{{ }}` usage

Reply:
I think I fixed that.